### ISSUE TITLE
maint-1.0 can't build with python>2.7

### ISSUE THREAD

**Issue by xylar** [E3SM-Project/E3SM#3304]
Two python scripts used in MPAS builds:
https://github.com/CVMix/CVMix-src/blob/master/src/shared/makedep.py
https://github.com/E3SM-Project/Ocean-BGC/blob/master/makedep.py
have been updated to python 3 in their respective branches but these changes have not made it to `maint-1.0`, causing tests to fail.

Given that build and testing environments are all moving to python 3, it will presumably be necessary to make a patch.

**Comment by xylar** [comment]
See https://github.com/E3SM-Project/E3SM_test_scripts/pull/13#pullrequestreview-314324671

**Comment by xylar** [comment]
@mark-petersen and @jonbob, I think you will have the best sense of how this patch can happen without bringing in other (possibly non-bit-for-bit) changes to `maint-1.0`.  How is the MPAS source patched for this branch in general?

**Comment by mark-petersen** [comment]
We have corresponding branches in MPAS-Dev. For example, we can merge bug fixes into e3sm/maint-1.0 on MPAS:
```
*   9b1c8e02 (HEAD -> e3sm/maint-1.0) Merge PR #103 into e3sm/maint-1.0
|\
| * bf6025bc Increase AM dimension from 32 to 36
|/
*   ceac0bc3 Merge PR #165 into e3sm/maint-1.0
|\
| * 4c61e096 Change start index from 1 to 2 on activeTracerVerticalAdvectionTopFlux
|/
*   95e121ab Merge PR #322 'captureEdgeTracerFluxes_maint1.0' into e3sm/maint-1.0
```
so we can branch off that and make a PR for the python updates, merge in, then update the E3SM maint-1.0 branch to the updated mpas-source commit.

**Comment by xylar** [comment]
That's great! The complication is that each of these files gets download automatically during MPAS build. The Ocean BGC fix should just be a question of updating the branch getting pulled down because no other changes have been made. The same is unfortunately not true of CVMix.

**Comment by mark-petersen** [comment]
@xylar I'm happy to update all the maintenance branches. If you have time to find the relevant PRs and commits from the last 4 months, that would be helpful. If not, I can do it too next week.

**Comment by xylar** [comment]
We would need the 2 commits from these two PRs:
https://github.com/MPAS-Dev/MPAS-Model/pull/274
https://github.com/MPAS-Dev/MPAS-Model/pull/304
Commits are:
6b1f556aa677b35dc9d226eca3fc224d3938a206
55a4f9c49b3de31512fa0b0fdf306aaddc54ebb1

The CVMix fix is more complicated.  We would need to start with:
https://github.com/CVMix/CVMix-src/tree/v0.84-beta
and cherry-pick onto that:
https://github.com/CVMix/CVMix-src/pull/76
Commit:
2c5842cc42b60a3ede681d1f62950e41bfeb969c
I'm not sure how we would do this.  One option would be to create a branch on someone's fork of CVMix and point the MPAS-Model maint-1.0 branch to this fork and branch instead of the main CVMix fork and the beta-8.4 tag.  This doesn't seem great but we don't really want to add an E3SM-specific branch or tag to the main CVMix repo and we won't currently have a mechanism I'm aware of for patching CVMix within MPAS (which I think we should have...).

**Comment by rljacob** [comment]
If you need to tie maint-1.0 to a fork of CVMix, use this one:  https://github.com/E3SM-Project/CVMix-src.

**Comment by xylar** [comment]
That's great!  If we have an E3SM fork of CVMix, that's clearly the right place to cherry-pick the commit I listed above and then use the resulting commit in maint-1.0.

**Comment by xylar** [comment]
I believe I have the necessary fixes in place.  We would need to first merge:
E3SM-Project/Ocean-BGC#3
E3SM-Project/CVMix-src#1
and then
MPAS-Dev/MPAS-Model#392
and finally update the submodule in E3SM-Project/E3SM/maint-1.0

@mark-petersen, it sounds like you should be able to help with that process and the necessary testing?

**Comment by xylar** [comment]
I believe this has been addressed on maint-1.0.  I am closing this but if I am wrong, please re-open.
